presto-main/src/test/java/com/facebook/presto/type/TestTimeBase.java

```diff
@@ -174,6 +179,36 @@ public void testCastToTimeWithTimeZone()
                 new SqlTimeWithTimeZone(new DateTime(1970, 1, 1, 3, 4, 5, 321, DATE_TIME_ZONE).getMillis(), DATE_TIME_ZONE.toTimeZone()));
     }
 
+    @Test
+    public void testCastToTimeWithTimeZoneWithTZWithRulesChanged()
+    {
+        TimeZoneKey timeZoneThatChangedSince1970 = getTimeZoneKey("Asia/Kathmandu");
+        DateTimeZone dateTimeZoneThatChangedSince1970 = getDateTimeZone(timeZoneThatChangedSince1970);
+        Session session = testSessionBuilder()
+                .setTimeZoneKey(timeZoneThatChangedSince1970)
+                .build();
+        FunctionAssertions localAssertions = new FunctionAssertions(session);
```

Comment:
`FunctionsAssertions` are closeable (and contain `LocalQueryRunner`) and should be closed.